keycloak_user_export/management/commands/migrate_users_to_keycloak.py

```diff
@@ -0,0 +1,224 @@
+                    user_keycloak_payload["credentials"].append(
+                        {
+                            "secretData": json.dumps(
+                                {"value": hash, "salt": base64_salt.decode()}
+                            ),
+                            "type": "password",
+                            "credentialData": json.dumps(
+                                {
+                                    "hashIterations": iterations,
+                                    "algorithm": "pbkdf2-sha256",
+                                }
+                            ),
+                        }
+                    )
+                unsynced_users_array.append(user_keycloak_payload)
+            headers = {
+                "Content-Type": "application/json",
+                "Authorization": f"Bearer {access_token}",
+            }
+            test_payload = json.dumps(
+                {
+                    "ifResourceExists": "SKIP",
+                    "realm": "master",
+                    "users": unsynced_users_array,
+                }
+            )
+            response = requests.request("POST", url, headers=headers, data=test_payload)
+            # If Keycloak responds with a 401, refresh the access_token and retry once.
+            if response.status_code == 401:
+                access_token = self.get_access_token(
+                    kwargs["client_id"],
+                    kwargs["email"],
+                    kwargs["password"],
+                    kwargs["client_secret"],
+                )
+                headers = {
+                    "Content-Type": "application/json",
+                    "Authorization": f"Bearer {access_token}",
+                }
+                response = requests.request(
+                    "POST", url, headers=headers, data=test_payload
+                )
+
+            # If the response from Keycloak is not 200, print out an error and move on to
+            # the next batch of users to export.
+            if response.status_code != 200:
+                print(
+                    f"Error calling Keycloak REST API, returned {response.status_code}"
+                )
+            else:
+                user_synced_with_keycloak_records = []
+                keycloak_response_body = json.loads(response.content)
+
+                # Expect the response below from Keycloak.
+                # {"overwritten":0,"added":0,"skipped":1,"results":[{"action":"ADDED","resourceType":"USER","resourceName":"collinp@mit.edu","id":"b2fb40bc-5c68-4f4b-b3ab-d178cd593454"}]}
+                keycloak_results = keycloak_response_body["results"]
+                for user in batch:
+                    keycloak_response_for_user = next(
+                        keycloak_result
+                        for keycloak_result in keycloak_results
+                        if keycloak_result["resourceName"] == user.email
+                    )
```

Comment:
Over the tens/hundreds of thousands this might add up to a lot of unnecessary looping over this list, I'd suggest mapping `keycloak_results` into a `dict` keyed off `resourceName` as a one-time cost per batch so that this lookup is O(1) instead of O(n).